sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1761,4 +1763,67 @@ private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
     }
     return supportedRunners.build();
   }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties.
+   */
+  public static PipelineOptions fromProperties(final Properties properties) {
+    if (properties == null) {
```

Comment:
Properties cannot be null. Our codebase has the policy that nothing is nullable unless explicitly marked. And here there is no benefit to doing that.

Reply:
Hmm, assuming this will be used under the hood and beam can't assume the properties instantiation/lookup I'd like to accept null here. Should I just add @Nullable?

Reply:
I don't understand your comment. I don't think it makes sense for any of these to accept `null`.

Reply:
It is about making like easy for end users.

Typical code will be:

    Properties p = loadCustomProps(); // can be null, no way beam can help here
    return fromProperties(p);

To avoid a null check in all user apps/lib I d prefer to put it in a single place, ie beam.